### PR TITLE
Add Initial Apache Camel Competitor page

### PR DESCRIPTION
**Pull request by xlight05**
## Purpose
$Subject

> Fixes https://github.com/ballerina-platform/ballerina-dev-website/issues/6909

## Checklist

- [ ] **Page addition**
  - [ ] Add `permalink` to pages.

- [ ] **Page removal**
  - [ ] Remove entry from corresponding left nav YAML file.
  - [ ] Add `redirect_from` on the alternative page.
  - [ ] If no alternative page, add redirection on the `redirections.js ` file.

- [ ] **Page rename**
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

- [ ] **Page restrcuture**
  - [ ] Add `permalink` to pages.
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

### CODE REVIEW

File: components/integration/replace-camel/code/camel-bbe/cloud-language.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+title: 'Language for the Cloud'
```
Comment (sm1990) [inline components/integration/replace-camel/code/camel-bbe/cloud-language.md:2]:
```suggestion
title: 'Language for the cloud'
```

File: components/integration/replace-camel/code/camel-bbe/connect-with-anything.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+title: 'Connect with anything'
+description: 'Ballerina has pre-built connectors and triggers for different protocols, 3rd party SaaS apps, etc., supporting various protocols, data formats, and authentication mechanisms. Ballerina Central, a globally hosted package management system, enables sharing and discovering packages, including those developed for unique organizational needs.'
```
Comment (sm1990) [inline components/integration/replace-camel/code/camel-bbe/connect-with-anything.md:3]: 'etc.' should be at the end of a list

File: components/integration/replace-camel/code/camel-bbe/data-manipulation.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+title: 'Powerful data manipulation'
+description: "Ballerina provides a seamless experience for data-intensive applications and integration scenarios. Effortlessly filter and extract data using query expressions, streamlining processing. Ballerina's powerful datamapper enables smooth data transformation, unlocking new possibilities for data manipulation. Unlock the power of data manipulation with Ballerina in your integration projects."
```
Comment (sm1990) [inline components/integration/replace-camel/code/camel-bbe/data-manipulation.md:3]:
```suggestion
description: "Ballerina provides a seamless experience for data-intensive applications and integration scenarios. Effortlessly filter and extract data using query expressions, streamlining processing. Ballerina's powerful data mapper enables smooth data transformation, unlocking new possibilities for data manipulation. Unlock the power of data manipulation with Ballerina in your integration projects."
```

File: components/integration/replace-camel/code/camel-bbe/diagram-when-you-need.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+title: "Flexible Development with Diagrams and Low-Code Capabilities"
```
Comment (sm1990) [inline components/integration/replace-camel/code/camel-bbe/diagram-when-you-need.md:2]:
```suggestion
title: "Flexible development with diagrams and low-code capabilities"
```

File: components/integration/replace-camel/code/camel-bbe/for-integrations.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
+---
+title: 'Language Built for Integrations'
+description: 'Ballerina excels with its first-class services, listeners, and clients, making integration capabilities central to its design. Build scalable integrations effortlessly, thanks to its strong type system and native support for JSON and XML. Experience the simplicity and readability of Ballerina for seamless integration development.'
+url: 'https://github.com/anupama-pathirage/ballerina-scenarios/tree/main/ballerina-zapier-samples/gdrive-new-event-to-slack-message'
```
Comment (sm1990) [inline components/integration/replace-camel/code/camel-bbe/for-integrations.md:4]: In other pages, the urls point to the main.bal
eg:- https://github.com/ballerina-platform/ballerina-dev-website/blob/master/components/ai/code/ai-bbe/create-ai-powered-apis.md

File: components/integration/replace-camel/code/camel-bbe/for-integrations.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    remote function onFolderTrash(drive:Change changeInfo) returns error? {
+    }
+}
+
+function transform(drive:Change change) returns slack:Message => {
+    channelName: slackChannel,
+    text: string `New file ${change.file?.name ?: ""} added at ${change.time ?: ""} .`
+};
```
[the remark this answers is not in the document]
Comment (sm1990) [reply components/integration/replace-camel/code/camel-bbe/for-integrations.md:49]: We usually add only the main function or the main service.

File: components/integration/replace-camel/code/camel-bbe/for-integrations.md (hunk 5 above)
Comment (sm1990) [reply components/integration/replace-camel/code/camel-bbe/for-integrations.md:4]: Samples should link to ballerina-guides

File: components/integration/replace-camel/code/camel-bbe/replace-dsl.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
+---
+title: 'Overcome DSL and XML Hurdles with Ease'
+description: 'Bid farewell to DSL complexities, XML, and workarounds. Embrace a programming language built for integrations, tailored for C-style language developers. With Ballerina, you can easily build custom components, unlocking even greater flexibility. Seamlessly build efficient integration solutions with confidence and ease, while experiencing effortless learning in the process.'
+url: 'https://github.com/anupama-pathirage/ballerina-scenarios/tree/main/ballerina-zapier-samples/github-new-issue-assigned-to-trello-card-and-twilio-sms'
```
Comment (sm1990) [inline components/integration/replace-camel/code/camel-bbe/replace-dsl.md:4]: Samples should link to ballerina-guides and should link to main.bal

File: components/integration/replace-camel/code/camel-bbe/replace-dsl.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+        string fileName = check file:basename(xmlFilePath);
+        if city == "London" {
+            log:printInfo("UK message");
+            check io:fileWriteXml(outputDirectory + "uk/" + fileName, message);
+        } else {
+            log:printInfo("Other message");
+            check io:fileWriteXml(outputDirectory + "others/" + fileName, message);
+        }
+    }
+}
```
Comment (sm1990) [inline components/integration/replace-camel/code/camel-bbe/replace-dsl.md:27]:
```suggestion
public function main() returns error? {
    file:MetaData[] inputDirectory = check file:readDir("data");
    string outputDirectory = "target/messages/";
    foreach file:MetaData readDirResult in inputDirectory {
        string xmlFilePath = readDirResult.absPath;
        xml message = check io:fileReadXml(xmlFilePath);
        string city = (message/**/<person>/**/<city>).data();
        string fileName = check file:basename(xmlFilePath);
        if city == "London" {
            log:printInfo("UK message");
            check io:fileWriteXml(outputDirectory + "uk/" + fileName, message);
        } else {
            log:printInfo("Other message");
            check io:fileWriteXml(outputDirectory + "others/" + fileName, message);
        }
    }
}
```

**Review by sm1990 — APPROVED** [review]

File: pages/usecases/integration/replace-camel/index.js [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+          name="description"
+          content="Write code with integration-friendly abstractions."
+        />
+        <meta name="author" content="WSO2 LLC" />
+        <meta
+          name="keywords"
+          content="ballerina, learn, documentation, docs, programming language"
+        />
+        <link rel="shortcut icon" href="/img/favicon.ico" />
+        <title>Next-Generation Integration: Elevating Integration to New Heights Beyond Apache Camel</title>
```
Comment (anupama-pathirage) [inline pages/usecases/integration/replace-camel/index.js:107]: Need to use sentence case for titles.

File: components/integration/replace-camel/intro/Intro.js [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+                    <Row className='pageContentRow integration'>
+                        <Col xs={12} lg={5} className={styles.introText}>
+                            <div>
+
+                                <p className={styles.introText2}>
+                                    Are DSL complexities and the steep learning curve of Apache Camel holding you back from seamless integration?
+                                </p>
+
+                                <p className={styles.introText3}>
+                                    Its time to break free and unlock the power of Ballerina. a purpose-built language designed to simplify integration challenges. With its elegant simplicity, modern architecture, and native support for cloud-native environments, Ballerina emerges as the preferred choice, paving the way for the future of integration with ease and efficiency.
```
Comment (anupama-pathirage) [inline components/integration/replace-camel/intro/Intro.js:41]:
```suggestion
                                    Its time to break free and unlock the power of Ballerina, a purpose-built language designed to simplify integration challenges. With its elegant simplicity, modern architecture, and native support for cloud-native environments, Ballerina emerges as the preferred choice, paving the way for the future of integration with ease and efficiency.
```

File: components/integration/replace-camel/code/camel-bbe/replace-dsl.md [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
```
Comment (anupama-pathirage) [inline components/integration/replace-camel/code/camel-bbe/replace-dsl.md:23]: Seems the code in sample link and this code snippet are not matching.

File: components/integration/replace-camel/code/camel-bbe/for-integrations.md [hunk 12]
```diff
@@ -0,0 +1,50 @@
+---
+title: 'Language built for integrations'
+description: 'Ballerina excels with its first-class services, listeners, and clients, making integration capabilities central to its design. Build scalable integrations effortlessly, thanks to its strong type system and native support for JSON and XML. Experience the simplicity and readability of Ballerina for seamless integration development.'
+url: 'https://github.com/anupama-pathirage/ballerina-scenarios/tree/main/ballerina-zapier-samples/gdrive-new-event-to-slack-message'
+---
+```
+import ballerinax/trigger.google.drive;
```
Comment (anupama-pathirage) [inline components/integration/replace-camel/code/camel-bbe/for-integrations.md:7]: I think we better use a different sample with a service, client and XML,json usage?

File: components/integration/replace-camel/code/camel-bbe/for-integrations.md (hunk 5 above)
Comment (xlight05) [reply components/integration/replace-camel/code/camel-bbe/for-integrations.md:4]: We'll point to bal files. We need to decide if we are gonna have the code for competitor scenarios in ballerina-guides tho

Comment (xlight05) [reply components/integration/replace-camel/code/camel-bbe/for-integrations.md:4]: We'll point to bal files. We need to decide if we are gonna have the code for competitor scenarios in ballerina-guides tho

**Comment by xlight05** [comment]
Addressed rest of the review comments in https://github.com/ballerina-platform/ballerina-dev-website/pull/7043

**Pull request merged**